src/flags/coord_flag_env_handler.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright 2024 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include <cstdint>
+#include <cstdlib>
+
+#include "flags/coord_flag_env_handler.hpp"
+#include "flags/coordination.hpp"
+#include "utils/logging.hpp"
+
+#include <spdlog/spdlog.h>
+
+namespace memgraph::flags {
+
+auto GetFinalCoordinationSetup() -> CoordinationSetup {
+  CoordinationSetup coordination_setup{};
+
+  auto *maybe_management_port = std::getenv(kMgManagementPort);
+  auto *maybe_coordinator_port = std::getenv(kMgCoordinatorPort);
+  auto *maybe_coordinator_id = std::getenv(kMgCoordinatorId);
+
+  bool are_envs_set = maybe_management_port || maybe_coordinator_port || maybe_coordinator_id;
+  bool are_flags_set = FLAGS_management_port || FLAGS_coordinator_port || FLAGS_coordinator_id;
+
+  if (are_envs_set) {
+    coordination_setup.management_port = maybe_management_port ? std::stoi(maybe_management_port) : 0;
```

Comment:
I think you can make yourself life easier by specifying constructor for CoordinatorSetup and then just passing one or other option

Reply:
this is simple logic but the code looks more complex than necessary IMO

Reply:
I agree that it looks complex, I think I fixed it now